ver_03/script.js

```diff
@@ -0,0 +1,306 @@
+
+  if (player1InputCheck == "") {
+    return "Please type <b><u>'play'</b></u> to roll your dices <br>and start the game. <br><br> Good Luck Player One!!";
+  }
+
+  if (player1InputCheck == "play") {
+    playerOneChoice = playerOneDiceRoll();
+  } else if (player1InputCheck == "keep") {
+    playerOneChoice = playerOneDiceDecisionK();
+  } else if (player1InputCheck == "swap") {
+    playerOneChoice = playerOneDiceDecisionS();
+  } else if (player1InputCheck == "result") {
+    playerOneChoice = compareResults();
+  }
+
+  return playerOneChoice;
+};
+
+var playerTwo = function (input2) {
+  player2InputCheck = input2.toLowerCase();
+
+  if (player2InputCheck == "") {
+    return "Please type <b><u>'play'</b></u> to roll your dices <br>and start the game. <br><br> Good Luck Player Two !! ";
+  }
+
+  if (player2InputCheck == "play") {
+    playerTwoChoice = playerTwoDiceRoll();
+  } else if (player2InputCheck == "keep") {
+    playerTwoChoice = playerTwoDiceDecisionK();
+  } else if (player2InputCheck == "swap") {
+    playerTwoChoice = playerTwoDiceDecisionS();
+  } else if (player2InputCheck == "result") {
+    playerTwoChoice = compareResults();
+  }
+
+  return playerTwoChoice;
+};
+
+// this is the dice rolling function returning number 1 - 6
+
+var playerOneDiceRoll = function () {
+  var diceRoll_1 = randomDiceRoll();
+  var diceRoll_2 = randomDiceRoll();
+
+  var playerOneChoice =
+    "Player One, you have rolled the following: " +
+    "<br><br>Dice one: " +
+    diceRoll_1 +
+    "<br> & " +
+    "<br>Dice two: " +
+    diceRoll_2 +
+    "<br><br> Would you like to keep or swap their positions? <br><br><br>Type <u><b>'keep'</b></u> to use the current configuration.<br><br>Type <u><b>'swap'</b></u> to change their positions";
+  playerOneRoll.push(diceRoll_1, diceRoll_2);
+  console.log(playerOneRoll);
+  return playerOneChoice;
+};
+
+var playerOneDiceDecisionK = function () {
+  var diceRoll_1 = playerOneRoll[0];
+  var diceRoll_2 = playerOneRoll[1];
+
+  var playerOneDecisionK =
+    "Player One, you have chosen to keep the dice." +
+    "<br><br> New dice one: " +
+    diceRoll_1 +
+    " <br>& " +
+    " <br>New dice two: " +
+    diceRoll_2 +
+    " <br><br>Please wait for Player Two to complete their turn before <br>typing <u><b>'result'</b></u> to get the comparison.";
+
+  return playerOneDecisionK;
+};
+var playerOneDiceDecisionS = function () {
+  var diceTemp = playerOneRoll[0];
+  playerOneRoll[0] = playerOneRoll[1];
+  playerOneRoll[1] = diceTemp;
+
+  var diceRoll_1 = playerOneRoll[0];
+  var diceRoll_2 = playerOneRoll[1];
+
+  var playerOneDecisionS =
+    "Player One, you have chosen to swap the dice." +
+    "<br><br>New dice one: " +
+    diceRoll_1 +
+    " <br>& " +
+    " <br>New dice two: " +
+    diceRoll_2 +
+    " <br><br>Please wait for Player Two to complete their turn before <br>typing <u><b>'result'</b></u> to get the comparison.";
+
+  return playerOneDecisionS;
+};
```

Comment:
We can merge this 2 functions into one that takes in an input `swap` or `keep`. That would have made the function more reusable across these 2 scenarios.